src/SAML2/Compat/Ssp/Logger.php

```diff
@@ -124,29 +125,35 @@ public function debug($message, array $context = array())
     public function log($level, $message, array $context = array())
     {
         switch ($level) {
-            case \SimpleSAML\Logger::ALERT:
-                \SimpleSAML\Logger::alert($message);
+            /* From PSR:  Calling this method with one of the log level constants
+            MUST have the same result as calling the level-specific method
+            */
+            case LogLevel::ALERT:
+                $this->alert($message, $context);
                 break;
-            case \SimpleSAML\Logger::CRIT:
-                \SimpleSAML\Logger::critical($message);
+            case LogLevel::CRITICAL:
+                $this->critical($message, $context);
                 break;
-            case \SimpleSAML\Logger::DEBUG:
-                \SimpleSAML\Logger::debug($message);
+            case LogLevel::DEBUG:
+                $this->debug($message, $context);
                 break;
-            case \SimpleSAML\Logger::EMERG:
-                \SimpleSAML\Logger::emergency($message);
+            case LogLevel::EMERGENCY:
+                $this->emergency($message, $context);
                 break;
-            case \SimpleSAML\Logger::ERR:
-                \SimpleSAML\Logger::error($message);
+            case LogLevel::ERROR:
+                $this->error($message, $context);
                 break;
-            case \SimpleSAML\Logger::INFO:
-                \SimpleSAML\Logger::info($message);
+            case LogLevel::INFO:
+                $this->info($message, $context);
                 break;
-            case \SimpleSAML\Logger::NOTICE:
-                \SimpleSAML\Logger::notice($message);
+            case LogLevel::NOTICE:
+                $this->notice($message, $context);
                 break;
-            case \SimpleSAML\Logger::WARNING:
-                \SimpleSAML\Logger::warning($message);
+            case LogLevel::WARNING:
+                $this->warning($message, $context);
+                break;
+            default:
+                throw new \Psr\Log\InvalidArgumentException("Unrecognized log level '$level''");
```

Comment:
This is the only doubt that I have about this PR. Could this lead to people getting exceptions all of a sudden, when before this PR they were only missing some log messages?

Reply:
yeah, I had some concerns about that.  The PSR says

> Calling this method with a level not defined by this specification MUST throw a Psr\Log\InvalidArgumentException if the implementation does not know about the level. Users SHOULD NOT use a custom level without knowing for sure the current implementation supports it.

I guess we can say this logger "knows" about any values of $level and make the `default` case log at the debug level. Do you think we should also make the `default` case log "unrecognized level $level" as well? or just log the message and context at debug level?  I don't have a preference - I'm using this with Guzzle which only uses the PSR levels so our choice doesn't impact me.

Reply:
Thinking about it twice, we always use the logger via the different _`loglevel`_`()` functions, so we should never reach the default anyway. Considering the PSR says explicitly that an exception should be raised, I think your code is perfectly reasonable like that 😉